shell/src/main/java/org/apache/kafka/shell/MetadataNodeManager.java

```diff
@@ -333,6 +339,44 @@ private void handleCommitImpl(MetadataRecordType type, ApiMessage message)
                 producerIds.create("nextBlockStartId").setContents(record.nextProducerId() + "");
                 break;
             }
+            case FEATURE_LEVEL_RECORD: {
+                FeatureLevelRecord record = (FeatureLevelRecord) message;
+                DirectoryNode featureLevels = data.root.mkdirs("featureLevels");
+                featureLevels.create(record.name()).setContents(Short.toString(record.featureLevel()));
+                break;
+            }
+            case BROKER_REGISTRATION_CHANGE_RECORD: {
+                BrokerRegistrationChangeRecord record = (BrokerRegistrationChangeRecord) message;
+                DirectoryNode brokerNode = data.root.mkdirs("brokers", Integer.toString(record.brokerId()));
+                FileNode registrationNode = brokerNode.create("registration");
+                JsonNode node = BrokerRegistrationChangeRecordJsonConverter.
+                        write(record, BrokerRegistrationChangeRecord.HIGHEST_SUPPORTED_VERSION);
+                registrationNode.setContents(node.toPrettyString());
+                if (record.fenced() == Byte.parseByte("1")) {
```

Comment:
We can use `BrokerRegistrationFencingChange.fromValue`

Reply:
Nice suggestion. Thanks.